unittests/test_cli.py

```diff
@@ -591,6 +591,42 @@ def test_unload_module(run_reframe, user_exec_ctx):
     assert returncode == 0
 
 
+def test_unuse_module_path(run_reframe, user_exec_ctx, monkeypatch):
+    ms = rt.runtime().modules_system
+    if ms.name == 'nomod':
+        pytest.skip('no modules system found')
+
+    # We need to have MODULEPATH set to unuse a module
+    monkeypatch.setenv('MODULEPATH', '')
+    module_path = os.path.abspath('unittests/modules')
+    returncode, stdout, stderr = run_reframe(
+        config_file=fixtures.USER_CONFIG_FILE,
+        more_options=[f'--unuse-module-path={module_path}'], action='run'
+    )
+
+    assert stdout != ''
+    assert 'Traceback' not in stdout
+    assert 'Traceback' not in stderr
+    assert returncode == 0
+
+
+def test_use_module_path(run_reframe, user_exec_ctx):
+    ms = rt.runtime().modules_system
+    if ms.name == 'nomod':
+        pytest.skip('no modules system found')
+
+    module_path = os.path.abspath('unittests/modules')
+    returncode, stdout, stderr = run_reframe(
+        config_file=fixtures.USER_CONFIG_FILE,
+        more_options=[f'--use-module-path={module_path}'], action='run'
```

Comment:
You should also try to load a module as well, otherwise you don't know that the module was loaded. You need to pass `-m testmod_foo`.

Reply:
And then you should check that there is not warning in the output (check the code the warning printeed when the requested module could not be loaded).